internal/internal_task_handlers.go

```diff
@@ -930,7 +961,17 @@ func (w *workflowExecutionContextImpl) retryLocalActivity(lar *localActivityResu
 	if backoff > 0 && backoff <= w.GetDecisionTimeout() {
 		// we need a local retry
 		time.AfterFunc(backoff, func() {
-			if _, ok := w.eventHandler.pendingLaTasks[lar.task.activityID]; !ok {
+			// TODO: this should not be a separate goroutine as it introduces race condition when accessing eventHandler.
+			// currently this is solved by changing eventHandler to an atomic.Value. Ideally, this retry timer should be
+			// part of the event loop for processing the workflow task.
+			eventHandler := w.getEventHandler()
+
+			// if decision heartbeat failed, the workflow execution context will be cleared and eventHandler will be nil
+			if eventHandler == nil {
+				return
+			}
+
+			if _, ok := eventHandler.pendingLaTasks[lar.task.activityID]; !ok {
```

Comment:
not related to your change - but looks like there could be a go-routine leak in laTunner.sendTask(lar.task) when localActivityTaskPoller shutsdown due to worker.Stop() - the fix would be to have sendTask select on {taskCh, stopC} - consider fixing this

Reply:
Makes sense. I will create another pr to address this issue.

Reply:
https://github.com/uber-go/cadence-client/pull/918